### PR TITLE
Replace the custom fingerprint auth UI with BiometricPrompt

### PR DESCRIPTION
**Pull request by alexbakker**
This patch replaces the usage of the deprecated FingerprintManager API with
BiometricPrompt. This uses the Android X library, so we get the native biometric
prompt on recent versions of Android and a Google-made one on older versions. By
not working with custom prompts for biometric authentication like we do now, we
can be sure that any issues like #70, #81, #237 are not actually our fault.
Here's what it looks like:

As a nice aside, this also adds support for the new facial recognition as an
authentication method on Pixel 4 phones.

### CODE REVIEW

**Comment by alexbakker** [comment]
This is ready for review.

File: app/src/main/java/com/beemdevelopment/aegis/ui/slides/CustomAuthenticatedSlide.java [hunk 1]
```diff
@@ -67,89 +44,61 @@ public int getCryptType() {
...
         _bgColor = color;
     }
 
+    public void showBiometricPrompt() {
+        BiometricSlotInitializer initializer = new BiometricSlotInitializer(this, new BiometricsListener());
+        BiometricPrompt.PromptInfo info = new BiometricPrompt.PromptInfo.Builder()
+                .setTitle("Test title")
+                .setDescription("Test description")
+                .setNegativeButtonText(getString(android.R.string.cancel))
+                .build();
```
Comment (michaelschattgen) [inline app/src/main/java/com/beemdevelopment/aegis/ui/slides/CustomAuthenticatedSlide.java]: Please change title and description.

File: app/src/main/java/com/beemdevelopment/aegis/ui/PreferencesFragment.java [hunk 2]
```diff
@@ -730,29 +738,26 @@ public void onException(Exception e) {
...
 
             saveDatabase();
             updateEncryptionPreferences();
         }
 
         @Override
-        public void onException(Exception e) {
-            Toast.makeText(getActivity(), getString(R.string.encryption_enable_fingerprint_error) + e.getMessage(), Toast.LENGTH_SHORT).show();
+        public void onSlotInitializationFailed(@NonNull CharSequence errString) {
+            Toast.makeText(getActivity(), getString(R.string.encryption_enable_biometrics_error) + errString, Toast.LENGTH_LONG).show();
```
Comment (michaelschattgen) [inline app/src/main/java/com/beemdevelopment/aegis/ui/PreferencesFragment.java]: Dismissing the prompt should not show an error

File: app/build.gradle [hunk 3]
```diff
@@ -67,6 +67,7 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'com.amulyakhare:com.amulyakhare.textdrawable:1.0.1'
     implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation "androidx.biometric:biometric:1.0.0-rc01"
```
Comment (michaelschattgen) [inline app/build.gradle]: It looks like there is a newer version, could you update this dependency as well?

**Comment by mbernier85** [comment]
Is there a reason why this PR was not reviewed, I got a new Pixel 4, and biometrics doesn't work, i guess this would solve the issue.

If there are still changes to be done, please put them here. I might be able to take a look if the original author of the PR can't do it.

**Comment by alexbakker** [comment]
This patch is done. It just needs the final approval from @michaelschattgen, but he has been very busy lately, so this'll have to wait for a little while longer.

**Comment by michaelschattgen** [comment]
I was indeed very busy with moving into new place.

I'm sorry for the delay. It's now merged and will be available in the next release.

**Pull request merged**